src/earn/poolInfo.ts

```diff
@@ -43,3 +47,51 @@ export async function fetchAavePoolInfo({
     throw err
   }
 }
+
+export async function fetchAaveRewards({
+  walletAddress,
+  assetAddress,
+  contractAddress,
+  networkId,
+  allTokens,
+}: {
+  walletAddress: Address
+  assetAddress: Address
+  contractAddress: Address
+  networkId: NetworkId
+  allTokens: TokenBalance[]
+}) {
+  try {
+    const network = networkIdToNetwork[networkId]
+    Logger.debug(TAG, 'Fetching Aave reward info', {
+      assetAddress,
+      contractAddress,
+      network,
+    })
+    const result = await publicClient[network].readContract({
+      abi: aaveIncentivesV3Abi,
+      address: contractAddress,
+      functionName: 'getAllUserRewards',
+      args: [[assetAddress], walletAddress],
```

Comment:
Is this nested array expected?

Reply:
yes, the first argument to the contract method is an array